src/main/scala/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.scala

```diff
@@ -0,0 +1,501 @@
+                               read_pairs: Long,
+                               cs_families: Long,
+                               ss_families: Long,
+                               ds_families: Long,
+                               ds_duplexes: Long,
+                               ds_fraction_duplexes: Double,
+                               ds_fraction_duplexes_ideal: Double) extends Metric
+
+  /**
+    * Metrics describing the set of observed UMI sequences and the frequency of their observations.  The UMI
+    * sequences reported may have been corrected using information within a double-stranded tag family.  For
+    * example if a tag family is comprised of three read pairs with UMIs ACGT-TGGT ACGT-TGGT ACGT-TGGG that
+    * a consensus UMI of will be generated ACGT-TGGT will be generated, and 3 raw observations counted for each
+    * of ACGT and TGGT, and no observations counted for TGGG.
+    *
+    * @param umi the possibly-corrected UMI sequence
+    * @param raw_observations the number of read pairs in the input BAM that observe the UMI (after correction)
+    * @param raw_observations_with_errors the subset of raw-observations that underwent any correction
+    * @param unique_observations the number of double-stranded tag families (i.e unique double-stranded molecules)
+    *                            that observed the UMI
+    * @param fraction_raw_observations the fraction of all raw observations that the UMI accounts for
+    * @param fraction_unique_observations the fraction of all unique observations that the UMI accounts for
+    */
+  case class UmiMetric(umi: String,
+                       var raw_observations: Long = 0,
+                       var raw_observations_with_errors: Long = 0,
+                       var unique_observations: Long = 0,
+                       var fraction_raw_observations: Double = 0,
+                       var fraction_unique_observations: Double = 0
+                      ) extends Metric
+}
+
+// TODOs:
+// TODO: Better minor breaks for log-axis plots
+
+
+@clp(group=ClpGroups.Umi, description=
+  """
+    |Collects a suite of metrics to QC duplex sequencing data.
+    |
+    |Usage to be finalized when tool is finalized, but to include:
+    |  - Emphasis tha the input must be the BAM produced by GroupReadsByUmi with strategy=paired
+    |  - Definition of CS, SS and DS families and how they relate
+    |  - Definition of "duplex" and how minAbReads and minBaReads relate to it
+    |  - 1-2 sentence summary for each the various metrics files produced
+    |
+    |For plots to be generated R must be installed and the ggplot2 package installed with suggested
+    |dependencies. Successfully executing the following in R will ensure a working installation:
+    |    install.packages("ggplot2", repos="http://cran.us.r-project.org", dependencies=TRUE)
+  """)
+class CollectDuplexSeqMetrics
+( @arg(flag="i", doc="Input BAM file generated by GroupReadByUmi.") val input: PathToBam,
```

Comment:
You could explicitly state the sort order in case someone has sorted it to `coordinate` and needs to get it back in the correct order ;p